src/query/formats/src/field_encoder/csv.rs

```diff
@@ -39,7 +38,7 @@ impl FieldEncoderCSV {
                 true_bytes: TRUE_BYTES_LOWER.as_bytes().to_vec(),
                 false_bytes: FALSE_BYTES_LOWER.as_bytes().to_vec(),
                 null_bytes: NULL_BYTES_ESCAPE.as_bytes().to_vec(),
-                nan_bytes: NAN_BYTES_SNAKE.as_bytes().to_vec(),
+                nan_bytes: options.stage.nan_display.as_bytes().to_vec(),
```

Comment:
Can we unify encode and only decode let the user choose?

Reply:
[clickhouse TSV may require `nan`,  we have to make clickhouse driver work](https://github.com/datafuselabs/databend/discussions/8934#discussioncomment-4241662)

Reply:
this is another issue, and need to make sure it is safe before changing it.

Reply:
> this is another issue, and need to make sure it is safe before changing it.

I meant the opposite in the discuss:  
 user no need choose decode (load) if  we accept all common reprs.
they may want to choose encode(unload) to fullfil requirement of another system.